front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -427,54 +430,58 @@ export function AgentMessage({
               identifier: `viz-${message.sId}-${lineStart}`,
             }}
             key={`viz-${message.sId}-${lineStart}`}
+            conversationId={conversationId}
+            agentConfigurationId={agentConfiguration.sId}
           />
         );
       },
     };
-  }, [message.sId, owner]);
+  }, [owner, conversationId, message.sId, agentConfiguration.sId]);
 
   return (
-    <ConversationMessage
-      owner={owner}
-      user={user}
-      conversationId={conversationId}
-      messageId={agentMessageToRender.sId}
-      pictureUrl={agentConfiguration.pictureUrl}
-      name={`@${agentConfiguration.name}`}
-      buttons={buttons}
-      avatarBusy={agentMessageToRender.status === "created"}
-      reactions={reactions}
-      enableEmojis={!hideReactions}
-      renderName={() => {
-        return (
-          <div className="flex flex-row items-center gap-2">
-            <div className="text-base font-medium">
-              {AssitantDetailViewLink(agentConfiguration)}
+    <AgentMessageContext.Provider value={{ isLastMessage: isLastMessage }}>
```

Comment:
The implementation we had first, was to leverage the existing `MarkDownContentContext` so we avoid nested below one more message.

Reply:
Nice 👍  didn't know that one